### ISSUE TITLE
Renderings start sticking when a transparent electron window loses and comes back into focus

### ISSUE THREAD

**Issue by bakkiraju** [electron/electron#8557]
* Electron version: 1.4.15
* Operating system: Mac OSX El capitan

### Expected behavior
Should not be seeing Ghost renderings in a transparent electron window when it goes out and comes back into focus

### Actual behavior
I am using jquery flipster plugin using a transparent electron window, I can do a carousel effect all day long without any problems as long as my electron transparent window is in focus but once I open some other app and comeback to my electron window the divs added to the flipster,start getting sticky and I see ghost images getting rendered. Trying to always keep electron window on the top doesnt help, de-focus and focus back causes rendering to stick

### How to reproduce
https://github.com/bakkiraju/flipster_electron

You can checkout, do npm install and npm start to run, you should see some divs with val and string, use mouse clicks to play with carousel, bring some other window into focus, go back to electron window and see what I am describing

**Comment by groundwater** [comment]
Thanks for the report @bakkiraju - we'll review the issue and get back to you within the next 7 days. If you're able to isolate the issue, we're always happy for the help.

**Comment by bakkiraju** [comment]
Thanks guys for all the help. Any insight into what might be causing the this issue. Also is there some guideline and an API to redraw / re-render contents of the window through electron API that I am missing ?. Looking at the behavior of this carousel plugin I feel that I should just call re-draw on content change/update event.

**Comment by zeke** [comment]
@bakkiraju I am experiencing a similar issue with [one of my apps](https://github.com/ummoji/ummoji-desktop), but am unable to consistently reproduce it:

<img width="723" alt="screen shot 2017-02-08 at 4 23 37 pm" src="https://cloud.githubusercontent.com/assets/2289/22763995/b7bbd55a-ee1c-11e6-9185-7b33c8271487.png">

I started cloning your repo but it's too big for the wifi in this coworking space! 📡 

I have a few suggestions:

1. Try updating to the latest beta version of Electron: `npm i -D electron@beta` (currently 1.6.0)

2. Try setting the `hasShadow` option to `false` when instantiating the `BrowserWindow` instance.

**Comment by bakkiraju** [comment]
electron 1.6.0 seems to fix the stickiness issue without setting hasShadow to false.

**Comment by bakkiraju** [comment]
Want to correct above stmt, hasShadow: false indeed helps

**Comment by zeke** [comment]
Glad that fixed it. Thanks for reporting back, @bakkiraju

**Comment by TomasHubelbauer** [comment]
As of Electron 31, the issue is still there without `hasShadow: false`. Maybe not in all cases it used to be in, but I do have a repro which still exhibits the behavior today: https://github.com/TomasHubelbauer/electron-lingering-shadow-repro. I've commented on https://github.com/electron/electron/issues/8847 which was the issue I came across first regarding this. I believe one of the window ghosting issues should be reopened as this issue is not fully resolved even in latest Electron.
